### PR TITLE
Add support for building/pushing images via GCB

### PR DESCRIPTION
**Pull request by spiffxp**
Shamelessly patterned after kubernetes-sigs/boskos with some mods:
- don't paramaterize go_version
- don't use _output/bin
- explicitly pass DOCKER_ env vars to images/build.sh
- add a test-e2e target

I also took the liberty of renaming cmd/controller to cmd/secret-sync-controller

ref: https://github.com/kubernetes-sigs/k8s-gsm-tools/issues/2

### CODE REVIEW

**Comment by b01901143** [comment]
I think I prefer to have one for the sync controller and one for the rotator.

**Comment by spiffxp** [comment]
Discussed offline
- we'd like to see an image per binary
- will be structured as `cmd/{binary}`

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+GCP_PROJECT ?= k8s-jkns-gke-soak
+
+CMDS = $(notdir $(shell find ./cmd/ -maxdepth 1 -type d | sort))
+
+.PHONY: all
+all: build
+
+.PHONY: $(CMDS)
+$(CMDS):
+	go build ./cmd/$@
```
Comment (b01901143) [inline Makefile:29]: Build failure due to existing folder with the same name of the desired binary

Comment (b01901143) [inline Makefile:29]: Build failure due to existing folder with the same name of the desired binary

Comment (b01901143) [reply Makefile:29]: open https://github.com/kubernetes-sigs/k8s-gsm-tools/pull/16 to fix.

**Pull request merged**